newIDE/app/src/CommandPalette/UseRefInitHook.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+import { useRef } from 'react';
+
+const useRefInit = <T>(init: () => T): T => {
+  const instanceRef = useRef(null);
+
+  let instance = instanceRef.current;
+  if (instance !== null) return instance;
+  // Lazy init
```

Comment:
Nit: most of the time, write full sentence in comments:

`// Lazily create the ref object.`